File: javascript/src/types/tools.ts
export interface WebSearchTool {
  name: "web_search";
  type?: "search";
  max_tokens?: number;
  max_tokens_per_page?: number;
}

export interface SmartWebSearchTool {
  name: "smart_web_search";
  type?: "search";
  max_tokens?: number;
  max_tokens_per_page?: number;
}

export interface LiteWebSearchTool {
  name: "lite_web_search";
  type?: "search";
  max_tokens?: number;
  max_tokens_per_page?: number;
}

export interface FetchURLTool {
  name: "fetch_url";
  type?: "url_reader";
}

export interface FunctionTool {
  type: "function";
  name: string;
  description?: string;
  parameters?: Record<string, unknown>;
  strict?: boolean;
}

export interface SkillTool {
  type: "skill";
  name: string;
  version?: string;
  arguments?: Record<string, unknown>;
}

export type Tool = WebSearchTool | SmartWebSearchTool | LiteWebSearchTool | FetchURLTool | FunctionTool | SkillTool;

export interface PublicTool {
  object: "tool";
  name: string;
  type?: string;
  description?: string;
  parameters?: Record<string, unknown>;
  max_tokens?: number;
  max_tokens_per_page?: number;
  version?: string;
}

export interface ListToolsResponse {
  object: "list";
  data: PublicTool[];
}

File: python/src/agent_api/types/tools.py
from __future__ import annotations

from typing import Any, Literal, TypedDict

class WebSearchTool(TypedDict, total=False):
    name: Literal["web_search"]
    type: Literal["search"]
    max_tokens: int
    max_tokens_per_page: int


class SmartWebSearchTool(TypedDict, total=False):
    name: Literal["smart_web_search"]
    type: Literal["search"]
    max_tokens: int
    max_tokens_per_page: int


class LiteWebSearchTool(TypedDict, total=False):
    name: Literal["lite_web_search"]
    type: Literal["search"]
    max_tokens: int
    max_tokens_per_page: int


class FetchURLTool(TypedDict, total=False):
    name: Literal["fetch_url"]
    type: Literal["url_reader"]


class FunctionTool(TypedDict, total=False):
    type: Literal["function"]
    name: str
    description: str
    parameters: dict[str, Any]
    strict: bool


class SkillTool(TypedDict, total=False):
    type: Literal["skill"]
    name: str
    version: str
    arguments: dict[str, Any]


Tool = WebSearchTool | SmartWebSearchTool | LiteWebSearchTool | FetchURLTool | FunctionTool | SkillTool
